UnityProject/Assets/Scripts/UI/Right click/RightClickableResult.cs

```diff
@@ -0,0 +1,91 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+/// <summary>
+/// Encapsulates a set of right click elements (RightClickableElements)
+/// generated by an IRightClickable, describing everything needed
+/// to display a right click options to the user and allow them to click on the options.
+///
+/// Allows customizing / overriding the normal appearance of the right click option.
+/// </summary>
+public class RightClickableResult
+{
+	private List<RightClickableElement> elements;
+
+	public RightClickableResult()
+	{
+		elements = new List<RightClickableElement>();
+	}
+
+	/// <summary>
+	/// Static factory, same as constructor, to make chaining look nicer.
+	/// </summary>
+	/// <returns></returns>
+	public static RightClickableResult Create()
+	{
+		return new RightClickableResult();
+	}
+
+	/// <summary>
+	/// Add an element whose appearance will be based on the
+	/// RightClickOption with the specified name. Shorthand to avoid having to specify
+	/// the full resource path of the RightClickOption or manually construct a
+	/// RightClickableElement.
+	/// </summary>
+	/// <param name="optionName">name of the option as it exists in the
+	/// Resources/ScriptableObjects/Interaction/RightclickOptions folder</param>
+	/// <param name="action">Action to invoke when the option is clicked.</param>
+	/// <param name="bgColorOverride">Color to use instead of the RightClickOption's normal
+	/// background color.</param>
+	/// <param name="spriteOverride">Sprite to use instead of the RightClickOption's normal sprite.</param>
+	/// <param name="nameOverride">Name to use instead of the RightClickOption's name</param>
+	/// <returns>this RightClickableResult, to support chaining</returns>
+	public RightClickableResult AddElement(string optionName, Action action,
```

Comment:
There is some debate if we really need the custom backgrounds

My arguments for are, not too much to implement, it makes the UI look nice Once you've implemented it and it was in the previous system

Example of how it would be used
like with the crummy hive mind mode There is, you could right click and the background could be a swirly thing for any hive mind related action

like if it's a massive pain to implement then yeah, you don't have to add it